src/System.ComponentModel.TypeConverter/tests/TypeConverterTestBase.cs

```diff
@@ -0,0 +1,325 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.Linq;
+using System.IO;
+using System.Runtime.Serialization;
+using System.Runtime.Serialization.Formatters.Binary;
+using Xunit;
+
+namespace System.ComponentModel.Tests
+{
+    public abstract class TypeConverterTestBase : RemoteExecutorTestBase
+    {
+        public abstract TypeConverter Converter { get; }
+        public virtual IEnumerable<ConvertTest> ConvertToTestData() => Enumerable.Empty<ConvertTest>();
+        public virtual IEnumerable<ConvertTest> ConvertFromTestData() => Enumerable.Empty<ConvertTest>();
+
+        public virtual bool PropertiesSupported { get; } = false;
+        public virtual bool StandardValuesSupported { get; } = false;
+        public virtual bool StandardValuesExclusive { get; } = false;
+
+        [Fact]
+        public void ConvertTo_DestinationType_Success()
+        {
+            Assert.All(ConvertToTestData(), convertTest =>
+            {
+                // We need to duplicate this test code as RemoteInvoke can't
+                // create "this" as the declaring type is an abstract class.
+                if (convertTest.RemoteInvokeCulture == null)
```

Comment:
Basically if I extracted this to a static method and call it from here and called it also from within the `RemoteInvoke`, I get a bunch of confusing errors for `MethodAccessException` as RemoteInvoke tries to create the abstract class rather than the derived type